azure-kusto-data/tests/test_kusto_client.py

```diff
@@ -341,43 +327,186 @@ def test_admin_then_query(self, mock_post):
+    def test_admin_then_query(self, mock_post):
+        """Tests admin then query."""
+        client = self._create_kusto_client()
+        query = ".show tables | project DatabaseName, TableName"
+        response = client.execute_mgmt("PythonTest", query)
+        self._assert_admin_then_query_response(response)
+
+    @patch("requests.Session.post", side_effect=mocked_requests_post)
+    def test_dynamic(self, mock_post):
+        """Tests dynamic responses."""
+        client = self._create_kusto_client()
+        query = """print dynamic(123), dynamic("123"), dynamic("test bad json"),"""
+        """ dynamic(null), dynamic('{"rowId":2,"arr":[0,2]}'), dynamic({"rowId":2,"arr":[0,2]})"""
+        row = client.execute_query("PythonTest", query).primary_results[0].rows[0]
+        self._assert_dynamic_response(row)
+
     @patch("requests.Session.post", side_effect=mocked_requests_post)
     def test_empty_result(self, mock_post):
         """Tests dynamic responses."""
-        client = KustoClient("https://somecluster.kusto.windows.net")
+        client = self._create_kusto_client()
         query = """print 'a' | take 0"""
         response = client.execute_query("PythonTest", query)
         assert response.primary_results[0]
 
     @patch("requests.Session.post", side_effect=mocked_requests_post)
     def test_null_values_in_data(self, mock_post):
         """Tests response with null values in non nullable column types"""
-        client = KustoClient("https://somecluster.kusto.windows.net")
+        client = self._create_kusto_client()
         query = "PrimaryResultName"
         response = client.execute_query("PythonTest", query)
 
         assert response is not None
+
+
+@pytest.mark.skipif(not async_installed, reason="requires async")
+class KustoClientTestsAsync(AsyncUnittest, KustoClientMixin):
+    """Tests class for KustoClient ASync API"""
+    @staticmethod
+    def _mock_callback(url, **kwargs):
+        body = json.dumps(mocked_requests_post(str(url), **kwargs).json())
+        return CallbackResult(status=200, body=body)
+
+    def _mock_query(self, mocked):
+        url = "{host}/v2/rest/query".format(host=self.HOST)
+        mocked.post(url, callback=self._mock_callback)
+
+    def _mock_mgmt(self, mocked):
+        url = "{host}/v1/rest/mgmt".format(host=self.HOST)
+        mocked.post(url, callback=self._mock_callback)
+
+    @aioresponses()
+    def test_sanity_query_async(self, mocked):
+        """Async version of self.test_sanity_query"""
+        self._mock_query(mocked)
+        client = self._create_kusto_client()
+        response = self.loop.run_until_complete(client.execute_query_async("PythonTest", "Deft"))
+        self._assert_sanity_query_response(response)
+
+    @aioresponses()
+    def test_sanity_control_command_async(self, mocked):
+        """Async version of self.test_sanity_control_command"""
+        self._mock_mgmt(mocked)
+        client = self._create_kusto_client()
```

Comment:
pytest (and me) are against test classes. you should refrain from using `self` in tests

Reply:
I agree that pytest is the better approach...

But the way I see it, we either change test_kusto_client to be written in pytest completely or keep it the way it is today.

My thought was that you won't want to see a lot of changes in the tests (to make sure everything runs exactly like before)